### PR TITLE
ci: adds docs link checker

### PR DESCRIPTION
**Pull request by achrinza**
Signed-off-by: Rifa Achrinza <25147899+achrinza@users.noreply.github.com>

## Checklist

- [x] DCO (Developer Certificate of Origin) [signed in all commits](https://loopback.io/doc/en/contrib/code-contrib.html)
- [ ] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: .github/workflows/check-docs-links.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Link Checker
+        id: lc
+        uses: peter-evans/link-checker@v1
+        with:
+          args: >
+            --concurrency 3
+            --document-root "https://loopback.io/"
```
Comment (dhmlau) [inline .github/workflows/check-docs-links.yaml:14]: do we need `--document-root` if we've specified `docs/**/*.md` below?

File: .github/workflows/check-docs-links.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+            --exclude "/127\.0\.0\.1|github\.com\/strongloop\/loopback-next\/issues|github\.com\/strongloop\/loopback-next\/pull/i"
```
Comment (dhmlau) [inline .github/workflows/check-docs-links.yaml:15]: probably might want to include `http://localhost:3000` as well.

**Comment by dhmlau** [comment]
@achrinza, going through the error list, there might be some urls that we might want to add on the exclude list, do you mind if I push changes in your PR to test it out? Or that's what you're planning to do?

File: .github/workflows/check-docs-links.yaml (hunk 1 above)
Comment (achrinza) [reply .github/workflows/check-docs-links.yaml:14]: `--document-root` is used for absolute paths. In previous commits, there were errors with some links that follow this style: `/docs/en/...` due to the document root not being set.

**Comment by dhmlau** [comment]
@achrinza, I submitted a PR to fix the broken link reported in the link checker: https://github.com/strongloop/loopback-next/pull/6220. 

What's left are:
- github links
   - e.g. source code and issue link. The link checker reports HTTP 429 `Too many requests` error. 
- API docs links
   - e.g. docs/site/Dynamic-models-repositories-controllers.md has a link `./apidocs/repository.html` 
- relative path to the docs web site
   - e.g. Loopback-application-layout-reference.md references the `.html` files inside the table.

File: .github/workflows/check-docs-links.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+on: [push, pull_request]
+name: Check docs links
+jobs:
+  check-docs-links:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```
Comment (bajtos) [inline .github/workflows/check-docs-links.yaml:7]: I think you need to add a step to build apidoc files first.

Quoting from the Action output:

```
docs/site/Executing-database-commands.md
ERROR	./apidocs/repository.defaultcrudrepository.execute.md
Stat docs/site/apidocs/repository.defaultcrudrepository.execute.md: no such file or directory
```

Here is the npm script to run:

```
npm run tsdocs
```

I am not sure what does it require:
- we may need to bootstrap all dependencies (run `npm install`)
- we may need to compile all packages (`npm run build`)

Perhaps @raymondfeng would know more?

**Comment by bajtos** [comment]
I like the idea of having an automated link checker :+1:

**Pull request closed without merging**